File: 0058-length-of-last-word/0058-length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
      s=s.trim(); // remove the white spaces from the word
      String [] arr = s.split(" "); //split the string into array on the basic of space
      String lastWord = arr[arr.length-1]; //get the last word of the array

      return lastWord.length();
      
    }
}

File: 0058-length-of-last-word/0058-length-of-last-word.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
    

    let length = s.length-1;
    console.log('s length',length)

    //remove the last space 

    while(length >= 0){
        if(s[length] == " "){
            --length;
        }else{
            break;
        }
    }


    // new length after the end space is removed
    let newLen = s.length-1;
    console.log("new string length after end space is removeed",newLen)
    let strLen = 0;

    for(let i=length;i>=0;i--){
        if(s[i] != " " ){
            strLen++;
        }else{
            break;
        }
    }

    return strLen;
};